core/core.go

```diff
@@ -156,7 +157,11 @@ func (c *Core) InsertChain(blocks types.Blocks) (int, error) {
 // procAppendQueue sorts the append queue and attempts to append
 func (c *Core) procAppendQueue() {
 
-	if c.procCounter > c_normalListProcCounter {
+	normalListProcCounter := c_normalListProcCounter
+	if len(c.appendQueue.Keys()) < c_appendQueueThreshold {
```

Comment:
Some appendQueues will be length 0 while still not nearly synced. If you haven't received blocks from peers on a zone.
Also will be 0 at the very beginning